### ISSUE TITLE
SingleDatePicker with numberOfMonths={1} not displaying month tab

### ISSUE THREAD

**Issue by 7091lapS** [airbnb/react-dates#166]
I tried to implement the SingleDatePicker wrapped in the same way found in the examples. This is the result: 
![screen shot 2016-11-07 at 8 17 55 pm](https://cloud.githubusercontent.com/assets/12876467/20072555/2648fe2c-a529-11e6-9b13-69a1a577efaa.png)

For some reason the current month tab seems to be positioned outside the container. The month previous to the current month (October) is rendered in the position of the current month (November) , and is hidden because of the numberOfMonths={1} prop, therefore the white space.

This is how I've implemented the wrapper:

```
import React from 'react';
import { SingleDatePicker } from 'react-dates'
import 'react-dates/lib/css/_datepicker.css'

class SingleDatePickerWrapper extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      focused: false,
      date: null,
    };

    this.onDateChange = this.onDateChange.bind(this);
    this.onFocusChange = this.onFocusChange.bind(this);
  }

  onDateChange(date) {
    this.setState({ date });
  }

  onFocusChange({ focused }) {
    this.setState({ focused });
  }

  render() {
    const { focused, date } = this.state;

    return (
      <SingleDatePicker
        {...this.props}
        id="date_input"
        date={date}
        focused={focused}
        onDateChange={this.onDateChange}
        onFocusChange={this.onFocusChange}
      />
    );
  }
}

export default SingleDatePickerWrapper;
```

And this is how I'm using it: 

```
<SingleDatePickerWrapper
              numberOfMonths={1}
 />
```

I'm not sure if I'm doing something wrong or if I need to pass some extra props through the wrapper in order for the component to render correctly, but I'm using pretty much unchanged from the SingleDatePicker example.
Any help would be greatly appreciated :)

**Comment by majapw** [comment]
Woah! That's really weird. Can you try to locate the `transition-container` dom element and see what styles are being applied to it? It should have an inline width and height. Maybe also the transform is being applied weirdly to the `CalendarMonthGrid` DOM element? Your wrapper code looks solid.

**Comment by 7091lapS** [comment]
The inline styles for `transition-container`are `"width: 318px; height: 292px;"` 
The `CalendarMonthGrid` element has `transform: translateX(0px);` @majapw

**Comment by 7091lapS** [comment]
@majapw Because I'm not using the scss files provided,
I resolved by overriding the generated css:

I overrode this: 
```
.CalendarMonthGrid--horizontal {
  position: absolute;
  left: 9px;
  width: 1200px; }
```
with this:
```
.CalendarMonthGrid--horizontal {
  position: absolute;
  left: 0 !important;
  width: auto !important; }
```

**Comment by majapw** [comment]
I am happy you got something working, but am still really confused as to what is happening given that the 1 month example works out of the box in the storybook. It'd be good to get to the bottom of this.

**Comment by amlivingamliving** [comment]
I'm having the same problem using an identical `SingleDatePickerWrapper` to @7091lapS 's. 

It seems the problem originates with the transform on the `CalendarMonthGrid` element. If I manually set `style="transform: translateX(-300px);"` on `CalendarMonthGrid` using the dev tools, the calendar is successfully realigned. The default value is `style="transform: translateX(0px);"`.

**Comment by majapw** [comment]
@amlivingamliving So `style="transform: translateX(0px);"` and what should be happening is that the first `CalendarMonth` is absolutely positioned. Is that happening?

**Comment by amlivingamliving** [comment]
@majapw Yes, the first `CalendarMonth` is absolutely positioned. I should also mention that if I do not specify `numberOfMonths={1}` on `SingleDatePickerWrapper` and use the default value instead, the `CalendarMonth` elements are still offset, this time by `150px`. So setting `style="transform: translateX(-150px);"` on `CalendarMonthGrid` realigns the calendar elements.

**Comment by amlivingamliving** [comment]
@majapw I believe #256 correctly identifies the source of the problem. The parent container needs to be `text-align: left;`. Thanks.

**Comment by eaharonoff** [comment]
Hi, 
I am using the `DayPickerRangeController` component and I am having the same issue. I cloned the repo and ran the script `yarn build:css` to retrieve the CSS and copied it into my project. I also tried all the above recommendations to no avail. Here is what my wrapper looks like:

```
import React, { Component } from 'react';
import moment from 'moment';
import { DayPickerRangeController } from 'react-dates';

import { START_DATE } from './constants';

if (process.env.BROWSER) {
  require('./styles/_datepicker.css');
}

class DatePickerWrapper extends Component {

  state = {
    errorMessages: [],
    startDate: null,
    endDate: null,
    focusedInput: START_DATE
  }

  _onDateChange = ({ startDate, endDate }) => {
    this.setState({ endDate: endDate });
  }

  _onFocusChange = (focusedInput) => {
    this.setState({
      focusedInput: !focusedInput ? START_DATE : focusedInput
    });
  }

  render() {
    const {
      startDate,
      endDate,
      focusedInput
    } = this.state;

          <DayPickerRangeController
            startDate={ startDate } 
            endDate={ endDate }
            onDatesChange={ this._onDateChange } 
            focusedInput={ focusedInput } 
            onFocusChange={ this._onFocusChange } 
            numberOfMonths={ 1 }
            initialVisibleMonth={ () => moment() }
          />
      </div>
    );
  }
}

export default DatePickerWrapper;

```

Any help would be much appreciated :)

**Comment by eaharonoff** [comment]
:/
